docs/development/private-keys/package-signing.md

```diff
@@ -0,0 +1,169 @@
+# Package Signing Key Generation and Handling
+
+Last updated 2024-07-03.
+
+## Owernship
+Certificate and key generation and handling is overseen by ALESCo.
+
+## Key Generation
+
+This is a slow and tedious process.  It is important that no corners are cut and the utmost care is taken validating security through every step of the process.
+
+### Prepare a Secure Environment
+
+Download latest AlmaLinux live ISO from `https://repo.almalinux.org/`.  Using default Gnome mini for this guide as to not have any 3rd-party binaries present (EPEL).
+
+In the folder where you download the ISO, look at the `CHECKSUM` and `CHECKSUM.asc` files.  First let's verify that it is properly signed and that its contents can be authenticated.
+
+```bash
+wget https://repo.almalinux.org/almalinux/9/live/x86_64/CHECKSUM
+wget https://repo.almalinux.org/almalinux/9/live/x86_64/CHECKSUM.asc
+wget https://repo.almalinux.org/almalinux/RPM-GPG-KEY-AlmaLinux-9
+gpg --import RPM-GPG-KEY-AlmaLinux-9
+gpg --verify CHECKSUM.asc CHECKSUM
+```
+
+Now verify the downloaded ISO's sha256 checksum against that contained in the `CHECKSUM` file.
+
+Copy the ISO file to a flash drive.
+
+`dd if=almalinux.iso of=/dev/sdX`
+
+Boot the image.  Connect to the internet temporarily to run a single command:
+
+```bash
+sudo dnf install pcsc-lite
+```
+
+#### Download GPG Public Keys of Recipients
+If you are doing key exports as part of this, grab the public keys of all recipients.
+
+These keys are accurate at the time of this writing.  Verify them prior to use, of course.
+```bash
+curl <key> | gpg --import
+```
+
+It may also be helpful to open a copy of this guide in the browser before disconnecting from the internet.
```

Comment:
I'd argue that you almost _certainly_ want this guide open during the process.

Reply:
Some masochists might be fine with manually typing it all :p